File: Day7/StrongNumber.cpp
#include<iostream>
using namespace std;

// function to calculate factorial  

int getFactorial(int n){
    int fact = 1;
    while(n != 0){
        fact *= n;
        n--;
    }
    return fact;
}

// function to check if a number is strong 

int getIsArmStrongNumber(int num){

    if(num <0){           
        return -1;
    }
    if(num ==  0){
        return 0;
    }
    int temp = num;
    int sum =0;
    while(num > 0){
        int digit = num % 10;
        sum += getFactorial(digit);
        num /= 10;
    }
    return temp == sum;
}
int main(){
    int num;
    cout<<"Enter the Number: ";
    cin>>num;
    int ans = getIsArmStrongNumber(num);
    if(ans== 1){
        cout<<num<<" is a Strong Number.";
    }else if(ans == 0){
        cout<<num<<" is not a Strong Number.";
    }else{
        cout<<"strong number are not defined for negative number";
    }
    return 0;
}

File: Day7/strong_number.py
""" Program for the Strong Number"""

def get_factorial(n):

    """function to calculate factorial """

    prod=1
    while n>0:
        prod *= n
        n = n-1
    return prod

def get_is_strong_number(num):

    """function to check if a number is strong ."""

    if num <0:
        return -1
    if num ==  0:
        return 0

    temp=num
    total_sum = 0
    while temp>0:
        digit = temp % 10
        total_sum +=  get_factorial(digit)
        temp //= 10
    return num == total_sum

number = int(input("Enter the number : "))
ans = get_is_strong_number(number)

if ans == 1:
    print(number,"is a Strong Number. ")
elif ans == 0:
    print(number,"is not a Strong Number.")
else:
    print("strong number are not defined for negative number")
